conda_package_streaming/package_streaming.py

```diff
@@ -125,9 +126,12 @@ def stream_conda_component(
 
         zf = zipfile.ZipFile(fileobj or filename)
         file_id, _, _ = os.path.basename(filename).rpartition(".")
+        # this substitution compensates for web downloads from anaconda.org having
+        # the platform as a prefix
+        file_id = re.sub("^(osx|linux|win|noarch)(-.+?)?_", "", file_id)
         component_name = f"{component}-{file_id}"
         component_filename = [
-            info for info in zf.infolist() if info.filename.startswith(component_name)
+            info for info in zf.infolist() if component_name in info.filename
```

Comment:
This change isn't necessary with the regex substitution, and I think I'll revert it. I was misguided in this, because the unwanted subdir is actually at the start of `file_id`, not the start of `component`, so this change didn't work.

Reply:
https://en.wikipedia.org/wiki/FILE_ID.DIZ